server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -685,10 +701,14 @@ static int getIndexNumberOfRoutingShards(Settings indexSettings, @Nullable Index
      * table based on the live nodes.
      */
     static ClusterState clusterStateCreateIndex(ClusterState currentState, Set<ClusterBlock> clusterBlocks, IndexMetadata indexMetadata,
-                                                BiFunction<ClusterState, String, ClusterState> rerouteRoutingTable) {
-        Metadata newMetadata = Metadata.builder(currentState.metadata())
-            .put(indexMetadata, false)
-            .build();
+                                                BiFunction<ClusterState, String, ClusterState> rerouteRoutingTable,
+                                                BiFunction<Metadata.Builder, IndexMetadata, Metadata.Builder> metadataTransaction) {
```

Comment:
Maybe rename `metadataTransaction` to `metadataTransformer`?
Since if provided the function will change the metadata builder.

Reply:
Ah, that is a better name 👍